### PR TITLE
Converting auth section Firebase Android codelab

### PR DESCRIPTION
**Pull request by racheldoshcollins**
The sections of the Firebase FriendlyEats Android codelab up through and including Authentication should be updated for the Kotlin version of this app. I'll add the rest of the codelab sections in another PR

### CODE REVIEW

**Comment by samtstern** [comment]
@racheldoshcollins looks like the auto-preview is working but if you want to get a local preview server running you can look at the little "devserver" I made for another codelab:
https://github.com/firebase/emulators-codelab/tree/master/scripts

The key is the `claat` tool although I added some JS on top to give me compile-on-save behavior.

File: build-android-start/app/src/main/java/com/google/firebase/codelab/friendlychat/MainActivity.kt [hunk 1]
```diff
@@ -44,6 +46,12 @@ class MainActivity : AppCompatActivity() {
         // Initialize Firebase Auth and check if the user is signed in
         // TODO: implement
 
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
```
Comment (samtstern) [inline build-android-start/app/src/main/java/com/google/firebase/codelab/friendlychat/MainActivity.kt:49]: Is this edit in the `start` folder intentional?

File: steps/index.lab.md [hunk 2]
```diff
@@ -215,123 +218,113 @@ Next we'll add some basic Firebase Authentication code to the app to detect user
...
 Next, implement the `onActivityResult` method to `SignInActivity` to handle the sign in result. If the result of the Google Sign-In was successful, use the account to authenticate with Firebase:
 
-**SignInActivity.java**
+**SignInActivity.kt**
 
 ```
-@Override
-public void onActivityResult(int requestCode, int resultCode, Intent data) {
-    super.onActivityResult(requestCode, resultCode, data);
+super.onActivityResult(requestCode, resultCode, data)
```
Comment (samtstern) [inline steps/index.lab.md:316]: nit: I think showing the function signature is helpful here, rather than just the function body.

**Review by samtstern — APPROVED** [review]

File: build-android-start/app/src/main/java/com/google/firebase/codelab/friendlychat/MainActivity.kt (hunk 1 above)
Comment (racheldoshcollins) [reply build-android-start/app/src/main/java/com/google/firebase/codelab/friendlychat/MainActivity.kt:49]: Yes. I shouldn't have removed it from the start folder - it's in the final code

**Review by samtstern — APPROVED** [review]

**Pull request merged**